snapcraft/storeapi/_store_client.py

```diff
@@ -0,0 +1,266 @@
+import hashlib
+import os
+import urllib.parse
+from time import sleep
+
+import pymacaroons
+import requests
+
+import snapcraft
+from snapcraft import config
+from snapcraft.internal.indicators import download_requests_stream
+
+from . import logger
+from . import _upload
+from . import constants
+from . import errors
+
+from ._sso_client import SSOClient
+from ._snap_index_client import SnapIndexClient
+from ._sca_client import SCAClient
+from ._up_down_client import UpDownClient
+
+
+class StoreClient():
+    """High-level client for the V2.0 API SCA resources."""
+
+    def __init__(self):
+        super().__init__()
+        self.conf = config.Config()
+        self.sso = SSOClient(self.conf)
+        self.cpi = SnapIndexClient(self.conf)
+        self.updown = UpDownClient(self.conf)
+        self.sca = SCAClient(self.conf)
+
+    def login(self, email, password, one_time_password=None, acls=None,
+              packages=None, channels=None, save=True):
```

Comment:
It seems the mypy annotions have gotten lost in the rewrite? ie. on master the function has this signature:

        def login(self, email: str, password: str, one_time_password: str = None,
              acls: Iterable[str] = None, channels: Iterable[str] = None,
              packages: Iterable[Dict[str, str]] = None,
              config_fd: TextIO = None, save: bool = True) -> None:

Can you please add these again? They help spot issues with different types, shadowed variables and such early as part of `./runtests.sh static`.

Reply:
Sorry I'm not sure what happened there as well. Changing it now.

Reply:
Would you mind also adding ` -> None` to __init__? I'm afraid I only spotted this now.